fix(types-and-typeclasses): Reword for clarity (#165)
learnyouahaskell/learnyouahaskell.github.io: source_md/types-and-typeclasses.md

`minBound` and `maxBound` are interesting because they have a type of `(Bounded a) => a`.
In a sense they are polymorphic constants.

> All tuples are also part of `Bounded` if the components within it are part of `Bounded`.

Reviewer: "components within" doesn't sound natural to my ear. How about

```suggestion
All tuples are also part of `Bounded` if its components are part of `Bounded` as well.
```

Reviewer 2: Alternative: All tuple types are instances of `Bounded` if all of their component types are instances of Bounded.

Reviewer 3: Some day, I'd like to improve the book's language around class instances. As it stands today, the book is littered with phrases like "part of <_class_>" and "member of the <_class_> class". That is not how type classes are usually talked about, and also suggests that type classes are closed (i.e. one could make a list of all instances, or establish that something is _not_ an instance).

For now, I prefer @ulysses4ever's suggestion for its simplicity and consistency with the rest of the book.
It should say _their_ instead of _its_ though: _all tuples_ is plural.
Changing _tuples_ &mapsto; _tuple types_ would be more correct, but the former is probably already understood well enough.

Author: Great - I've made the adjustments.
